lib/hotel.rb

```diff
@@ -0,0 +1,56 @@
+require_relative 'room'
+require_relative 'block'
+
+module Hotel_System
+  class Hotel
+    attr_reader :number_of_rooms, :rooms
+    attr_accessor :blocks
+    def initialize(number_of_rooms)
+      @number_of_rooms = number_of_rooms
+      @rooms = []
+      number_of_rooms.times do |i|
+        @rooms << Hotel_System::Room.new(i + 1)
+      end
+      @blocks = []
+    end
+
+    def reservations
+      reservations = []
+      @rooms.each do |room|
+        if room.reservations != []
+          reservations += room.reservations
+        end
+      end
+      return reservations
+    end
+
+    def find_res_by_date(date)
+      all_reservations = self.reservations
+      date_reservations = all_reservations.select do |reservation|
+        (reservation.check_in...reservation.check_out).cover?(date)
```

Comment:
Not a big deal at all because I think it's fairly contained in this project, BUT, there could be some refactoring with this line.

Hotel is reaching into each reservation and performing the logic on if the reservation is in this date. It might be interesting to pull this into a small helper function so that reservation does it itself, so that in hotel it could be:
```
reservation.covers?(date)
```
and in `Reservation` there's a method
```
def covers? date
  (@check_in...@check_out).cover?(date)
end
```